plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt

```diff
@@ -354,43 +362,64 @@ open class KtlintPlugin : Plugin<Project> {
         reporters.set(extension.reporters)
     }
 
-    private fun Project.getMetaKtlintCheckTask(): Task = this.tasks.findByName(CHECK_PARENT_TASK_NAME)
-        ?: this.task(CHECK_PARENT_TASK_NAME).apply {
+    private fun Project.getMetaKtlintCheckTask(): TaskProvider<Task> = try {
+        this.tasks.named(CHECK_PARENT_TASK_NAME)
+    } catch (ignored: UnknownDomainObjectException) {
+        taskHelper(CHECK_PARENT_TASK_NAME) {
             group = VERIFICATION_GROUP
             description = "Runs ktlint on all kotlin sources in this project."
         }
+    }
```

Comment:
What's going on here? I don't understand.

Reply:
I try to get lazy task and if it fails to find - create a new one.

Reply:
Is there a way to do this without an exception being caught as that's expensive?
Also, if it can't be simplified, can you generify this bit of `try` & `catch` logic to be a generic method?